inngest/src/main/kotlin/com/inngest/Introspection.kt

```diff
@@ -0,0 +1,44 @@
+package com.inngest
+
+import com.beust.klaxon.Json
+
+abstract class Introspection(
+    open val functionCount: Int,
+    open val hasEventKey: Boolean,
+    open val hasSigningKey: Boolean,
+    open val mode: String,
+    @Json("authentication_succeeded") open val authenticationSucceeded: Boolean?,
+    @Json("schema_version") val schemaVersion: String = "2024-05-24",
+)
+
+internal data class InsecureIntrospection(
```

Comment:
any reason for using insecure/secure as the terminology when the spec and JS SDK are using unauthenticated/authenticated?

Reply:
i started implementing closely with the [Go sdk](https://github.com/inngest/inngestgo/blob/9e60ab4b900e11a823e0a51bd4ee4dc6d58d73be/handler.go#L639-L651), i can rename them since the spec and both JS and Python has it as unauthenticated/authenticated.

Reply:
Ah ok, fine to leave as is if there's precedent in the go sdk